src/PostgREST/Query.hs

```diff
@@ -273,3 +276,13 @@ runPreReq conf = lift $ traverse_ (SQL.statement mempty . stmt) (configDbPreRequ
       ("select " <> fromQi req <> "()")
       HD.noResult
       (configDbPreparedStatements conf)
+
+-- | Sets Timezone from prefer header
+setTimezone :: TimezoneNames -> Preferences -> DbHandler ()
+setTimezone allowedTimezones preferences = 
+  lift $
+    when (isJust $ L.find (== timezone) allowedTimezones)
+    $ do
+      SQL.sql $ "set local time zone '" <> T.encodeUtf8 timezone <> "'"
```

Comment:
To avoid another query (another roundtrip from PostgREST to the db), this can be on [setPgLocals](https://github.com/PostgREST/postgrest/blob/1e0b205e01312b191959a70bd63eb8089c2fe0a3/src/PostgREST/Query.hs#L238-L240). As:

```sql
select set_config('timezone', 'Europe/Berlin', true);
```